src/datanode/src/datanode.rs

```diff
@@ -506,6 +481,66 @@ impl DatanodeBuilder {
     }
 }
 
+/// Open all regions belong to this datanode.
+async fn open_all_regions(
+    region_server: RegionServer,
+    kv_backend: KvBackendRef,
+    open_with_writable: bool,
+    node_id: u64,
+) -> Result<()> {
+    let datanode_table_manager = DatanodeTableManager::new(kv_backend.clone());
+    let mut regions = vec![];
+    let mut table_values = datanode_table_manager.tables(node_id);
+
+    while let Some(table_value) = table_values.next().await {
```

Comment:
Is fetching to-be-opened regions suitable placed here if this function can be executed in background? I mean, if here goes wrong, it's pointless to do any background region open jobs.

Reply:
Sorry, I didn't get it. Do you meaning that we should move it out of the background task?

Reply:
yes